okio-files/src/commonTest/kotlin/okio/files/FileSystemTest.kt

```diff
@@ -68,4 +70,50 @@ class FileSystemTest {
         |""".trimMargin(), buffer.readUtf8())
     source.close()
   }
+
+  @Test
+  fun `file sink`() {
+    val path = "/tmp/FileSystemTest-file_sink.txt".toPath()
+    val sink = Filesystem.SYSTEM.sink(path)
+    val buffer = Buffer().writeUtf8("hello, world!")
+    sink.write(buffer, buffer.size)
+    sink.close()
+    assertTrue(path in Filesystem.SYSTEM.list("/tmp".toPath()))
+    assertEquals(0, buffer.size)
+    assertEquals("hello, world!", path.readUtf8())
+  }
+
+  @Test
+  fun `file sink flush`() {
+    val path = "/tmp/FileSystemTest-file_sink.txt".toPath()
+    val sink = Filesystem.SYSTEM.sink(path)
+
+    val buffer = Buffer().writeUtf8("hello,")
+    sink.write(buffer, buffer.size)
+    sink.flush()
+    assertEquals("hello,", path.readUtf8())
+
+    buffer.writeUtf8(" world!")
+    sink.write(buffer, buffer.size)
+    sink.close()
+    assertEquals("hello, world!", path.readUtf8())
+  }
+
+  @Test
+  fun `file sink no such directory`() {
+    try {
```

Comment:
assertFailsWith

Reply:
Is this in multiplatform? I couldn't find it

Reply:
Yep! https://github.com/square/okio/pull/798/